Sources/SwiftCoAP/SCMessage.swift

```diff
@@ -390,6 +364,16 @@ extension SCCoAPUDPTransportLayer: SCCoAPTransportLayerProtocol {
         semaphore.wait()
         return tlsOptions
     }
+    
+    private func cancelConnection(to endpoint: NWEndpoint) {
+        operationsQueue.sync { [weak self] in
```

Comment:
Haven't asked last time. Just wonder why do you feel `.sync` is better here? Can't it be a little notch faster to run it `.async` with a barrier? (Not sure and haven't measured though.)

Reply:
I'm a big fan of avoiding multithreading and async as much as possible. It makes reasoning and following up the logic simpler and in the end helps to avoid bugs.

In this specific case we do UDP message sending (happens instantly) or operations on the dictionary (also instant), so until we see issues with delays on the main thread, I prefer to keep it simple